src/lib.rs

```diff
@@ -292,44 +295,89 @@ impl Wallet {
     }
 }
 
-pub struct ExtendedKeyInfo {
+pub struct DescriptorKeyInfo {
+    xprv: Option<String>,
+    xpub: String,
+}
+
+struct ExtendedKeyInfo {
     mnemonic: String,
     xprv: String,
     fingerprint: String,
 }
 
-fn generate_extended_key(
-    network: Network,
-    word_count: WordCount,
-    password: Option<String>,
-) -> Result<ExtendedKeyInfo, Error> {
-    let mnemonic: GeneratedKey<_, BareCtx> =
+impl ExtendedKeyInfo {
+    fn new(
+        network: Network,
+        word_count: WordCount,
+        password: Option<String>,
+    ) -> Self {
```

Comment:
Can we retain `Result<T, E>` semantics here? Did we remove it intentionally? I'd prefer not having `.unwrap()` and `throw`ing exceptions in Kotlin etc.

Reply:
Got it, will add back the `Result<T, E>` and force push the changes.